multitenancy.md

```diff
@@ -0,0 +1,94 @@
+- `mtt-{1,2}` will be repositories from where we'll install workload clusters.
+
+Simply do a _search and replace_ if you want to name your clusters something else.
+`mtt` - simply stands for _multi-tenancy-test_ :)
+
+### Infrastructure and kubernetes
+1. Set some variables
+  ```bash
+  domain=<your_domain>
+  pgp_fp=<your_pgp_fp>
+  config_path=<path_to_store_config_repos>
+  ```
+
+1. Init repos
+  ```bash
+  ./dist/ck8s_linux_amd64 init mtt-0 exoscale --flavor dev --pgp-fp ${pgp_fp} --config-path ${config_path}/mtt-0
+  ./dist/ck8s_linux_amd64 init mtt-1 exoscale --flavor dev --pgp-fp ${pgp_fp} --config-path ${config_path}/mtt-1
+  ./dist/ck8s_linux_amd64 init mtt-2 exoscale --flavor dev --pgp-fp ${pgp_fp} --config-path ${config_path}/mtt-2
+  ```
+
+2. For each repo edit the configuration files
+
+3. Create infrastructure and install kubernetes for the clusters
+  ```bash
+  ./dist/ck8s_linux_amd64 apply --cluster sc --config-path ${config_path}/mtt-0
+  ./dist/ck8s_linux_amd64 apply --cluster wc --config-path ${config_path}/mtt-1
+  ./dist/ck8s_linux_amd64 apply --cluster wc --config-path ${config_path}/mtt-2
+  ```
+
+3. Get ip of sc loadbalancer
+  ```bash
+  sc_lb_ip=$(./dist/ck8s_linux_amd64 internal terraform output --cluster sc --config-path ${config_path}/mtt-0 | jq -r '.sc_ingress_controller_lb_ip_address.value')
+  ```
+
+4. Create DNS records
+  ```bash
+  exo dns add A ${domain} -a ${sc_lb_ip} -n *.ops.mtt-0
+  exo dns add A ${domain} -a ${sc_lb_ip} -n *.mtt-0
+  ```
+
+### Apps
+1. Init config
+  ```bash
+  export CK8S_CLOUD_PROVIDER=exoscale
+
+  export CK8S_ENVIRONMENT_NAME=mtt-0
+  export CK8S_CONFIG_PATH=${config_path}/mtt-0
+  ./bin/ck8s init
+
+  export CK8S_ENVIRONMENT_NAME=mtt-1
+  export CK8S_CONFIG_PATH=${config_path}/mtt-1
+  ./bin/ck8s init
+
+  export CK8S_ENVIRONMENT_NAME=mtt-2
+  export CK8S_CONFIG_PATH=${config_path}/mtt-2
+  ./bin/ck8s init
+  ```
+
+2. Edit
+  - `secrets.yaml` for all clusters
+  - `sc-config.yaml` for the service cluster
+  - `wc-config.yaml` for all workload cluster
+  You should set the same `opsDomain` and `baseDomain` in all clusters.
+
+### Teardown
+
+```bash
+# Tear down infra and k8s
+./dist/ck8s_linux_amd64 destroy --cluster sc --config-path ${config_path}/mtt-0 --kubernetes-cleanup=false
+./dist/ck8s_linux_amd64 destroy --cluster wc --config-path ${config_path}/mtt-1 --kubernetes-cleanup=false
+./dist/ck8s_linux_amd64 destroy --cluster wc --config-path ${config_path}/mtt-2 --kubernetes-cleanup=false
+
+# Remove dns records
+exo dns remove a1ck.io *.ops.mtt-0
+exo dns remove a1ck.io *.mtt-0
+```
+
+## Unknowns and failuers
+
+- OIDC and kubelogin has not been tested.
+
+- The certificate for the kubeapi-metrics ingress will never be ready in the workload clusters as all clusters share the same `baseDomain` and `opsDomain` which are both pointing to the service cluster.
+
+- Have not used ingress to any of the workload clusters.
```

Comment:
Can you use the same tense as above? "Ingress of workload clusters has not been tested, but is expected to work without issues."